judge/views/problem.py

```diff
@@ -219,9 +219,70 @@ def get_context_data(self, **kwargs):
                                           context['description'], 'problem')
         context['meta_description'] = self.object.summary or metadata[0]
         context['og_image'] = self.object.og_image or metadata[1]
+
+        context['can_vote'] = self.object.can_vote(user)
+        # The vote this user has already cast on this problem.
+        if context['can_vote']:
+            try:
+                vote = ProblemPointsVote.objects.get(voter=user.profile, problem=self.object)
+            except ObjectDoesNotExist:
+                vote = None
+            context['has_voted'] = vote is not None
+        else:
+            context['has_voted'] = False
+
+        if context['has_voted']:
+            context['voted_points'] = vote.points  # The previous vote's points.
+            context['voted_note'] = vote.note
+
+        all_votes = list(self.object.problem_points_votes.order_by('points').values_list('points', flat=True))
+
+        context['has_votes'] = len(all_votes) > 0
+
+        # If the user is not currently in contest and is authenticated.
+        if not user.is_authenticated or user.profile.current_contest is None:
```

Comment:
Condition backwards?

Reply:
Yea a bit yikes for this to slip pass me. I think its fixed now, it checks if user is authenticated and if current contest is None.